models/POP/shell_scripts/cesm2_1/assimilate.csh

```diff
@@ -0,0 +1,358 @@
+
+      (ls -rt1 ${CASE}.pop.output_postinf_sd.* | tail -n 1 >! latestfile) > & /dev/null
+      set nfiles = `cat latestfile | wc -l`
+
+      if ( $nfiles > 0 ) then
+         set latest = `cat latestfile`
+         ${LINK} $latest input_postinf_sd.nc
+      else
+         echo "ERROR: Requested POSTERIOR inflation but specified no incoming inflation SD file."
+         echo "ERROR: expected something like ${CASE}.pop.output_postinf_sd.YYYY-MM-DD-SSSSS.nc"
+         exit 2
+      endif
+
+   endif
+
+else
+   echo "Posterior Inflation       not requested for this assimilation."
+endif
+
+#=========================================================================
+# Block 5: REQUIRED DART namelist settings
+#=========================================================================
+${REMOVE} restarts_in.txt restarts_out.txt
+
+foreach FILE ( rpointer.ocn_????.restart )
+   head -n 1 ${FILE} >> restarts_in.txt
+end
+
+# WARNING: this is the part where the files just get overwritten
+${COPY} restarts_in.txt restarts_out.txt
+
+# POP always needs a pop_in and a pop.r.nc to start.
+# Lots of ways to get the filename
+
+set OCN_RESTART_FILENAME = `head -n 1 rpointer.ocn_0001.restart`
+
+${LINK} $OCN_RESTART_FILENAME pop.r.nc
+${LINK} pop_in_0001          pop_in
+
+#=========================================================================
+# Block 6: Actually run the assimilation. 
+# WARNING: this version just overwrites the input - no ability to recover
+#
+# DART namelist settings required:
+# &filter_nml:           async                    = 0,
+# &filter_nml:           adv_ens_command          = "no_CESM_advance_script",
+# &filter_nml:           input_state_file_list    = "restarts_in.txt"
+# &filter_nml:           output_state_file_list   = "restarts_out.txt"
+# &filter_nml:           stages_to_write          = 'preassim', 'output'
+# &filter_nml:           output_restarts          = .true.
+# &filter_nml:           output_mean              = .true.
+# &filter_nml:           output_sd                = .true.
+# &filter_nml:           write_all_stages_at_end  = .true.
+#
+# &filter_nml:           obs_sequence_in_name    = 'obs_seq.out'
+# &filter_nml:           obs_sequence_out_name   = 'obs_seq.final'
+# &filter_nml:           init_time_days          = -1,
+# &filter_nml:           init_time_seconds       = -1,
+# &filter_nml:           first_obs_days          = -1,
+# &filter_nml:           first_obs_seconds       = -1,
+# &filter_nml:           last_obs_days           = -1,
+# &filter_nml:           last_obs_seconds        = -1,
+# &ensemble_manager_nml: single_restart_file_in  = .false.
+# &ensemble_manager_nml: single_restart_file_out = .false.
+#
+#=========================================================================
+
+echo "`date` -- BEGIN FILTER"
+${LAUNCHCMD} ${EXEROOT}/filter || exit -6
+echo "`date` -- END FILTER"
+
+if ( $?LSB_PJL_TASK_GEOMETRY ) then
+   setenv LSB_PJL_TASK_GEOMETRY "${ORIGINAL_LAYOUT}"
+endif
```

Comment:
I believe this is detritus. At one point filter was running better on a subset of tasks (LSB_PJL_TASK_GEOMETRY)
than the whole processor set (there just wasn't enought obs to warrant it). This block reset the tasks to the original task list ... but I don't see the original list being created anywhere. Am I missing that?